From 6fc6d0f0237bc4dcbac470df975634da33104756 Mon Sep 17 00:00:00 2001
From: Peter Collingbourne <pcc@google.com>
Date: Fri, 16 Nov 2018 18:06:59 -0800
Subject: [PATCH 14/18] hwui: Fix a use-after-free found by HWAsan

The variable sizeResult is an iterator pointing into mCurrentValues,
and clearing mCurrentValues invalidates the iterator.

Bug: 119773465
Change-Id: I940ea397af87561d33f74a9d52abc3afd0a7538f
---
 libs/hwui/pipeline/skia/SkiaMemoryTracer.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libs/hwui/pipeline/skia/SkiaMemoryTracer.cpp b/libs/hwui/pipeline/skia/SkiaMemoryTracer.cpp
index ee996224fee6..ea578cb3ec05 100644
--- a/libs/hwui/pipeline/skia/SkiaMemoryTracer.cpp
+++ b/libs/hwui/pipeline/skia/SkiaMemoryTracer.cpp
@@ -106,8 +106,9 @@ void SkiaMemoryTracer::processElement() {
                 resourceValues.insert({key, sizeResult->second});
             }
         } else {
+            TraceValue sizeValue = sizeResult->second;
             mCurrentValues.clear();
-            mCurrentValues.insert({key, sizeResult->second});
+            mCurrentValues.insert({key, sizeValue});
             mResults.insert({resourceName, mCurrentValues});
         }
     }
-- 
2.17.1

